modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/serializer/RecordV2Serializer.java

```diff
@@ -266,7 +265,7 @@ private static WALPointer readPositionAndCheckPoint(
         int fileOff = in.readInt();
         int len = in.readInt();
 
-        if (!F.eq(idx, expPtr.index()) || (!skipPositionCheck && !F.eq(fileOff, expPtr.fileOffset())))
+        if (expPtr != null && (!F.eq(idx, expPtr.index()) || (!skipPositionCheck && !F.eq(fileOff, expPtr.fileOffset()))))
```

Comment:
Do we still need this check?

Reply:
In ByteBufferWalIterator we pass null as expPtr and that's why the pointer check is not executed here. I think this is the minimum change that we can do in the serializer class.

Reply:
Reverted this change. Also changed ByteBufferWalIterator so that it expects segment and offset.